File: LinkedTreeMapManager.java
package Second_sem.lab5;

import com.google.gson.internal.LinkedTreeMap;

public class LinkedTreeMapManager {

    private static String[] fieldNames = {"id", "name", "coordinates", "creationDate",
            "realHero", "hasToothpick", "impactSpeed", "soundtrackName",
            "minutesOfWaiting", "mood", "car"};


    public static LinkedTreeMap<String, Object> makeLinkedTreeMap(Object[] params){
        LinkedTreeMap<String, Object> linkedTreeMap = new LinkedTreeMap<>();
        for(int i = 0; i<fieldNames.length;i++){
            linkedTreeMap.put(fieldNames[i], params[i]);
        }
        return linkedTreeMap;
    }

    public static Object[] getParams(LinkedTreeMap<String, Object> linkedTreeMap){
        Object[] array = new Object[linkedTreeMap.size()];
        for(int i = 0; i< linkedTreeMap.size(); i++){
            array[i] = linkedTreeMap.get(fieldNames[i]);
        }
        return array;
    }

}

File: Kotlin/HelpingOrFormatingClasses/LinkedTreeMapManager.kt
package Second_sem.lab5.Kotlin.HelpingOrFormatingClasses

import com.google.gson.internal.LinkedTreeMap

val fieldNames = listOf<String>("id", "name", "coordinates", "creationDate",
    "realHero", "hasToothpick", "impactSpeed", "soundtrackName",
    "minutesOfWaiting", "mood", "car")


fun makeLinkedTreeMap(params: List<Any?>) : LinkedTreeMap<String, Any?> {
    val linkedTreeMap = LinkedTreeMap<String, Any?>()
    for (i in fieldNames.indices){
        linkedTreeMap.put(fieldNames[i], params[i])
    }
    return linkedTreeMap
}

fun getPatams(linkedTreeMap: LinkedTreeMap<String, Any?>) : List<Any?>{
    val array = mutableListOf<Any?>(linkedTreeMap.size)
    for(i in 0 until linkedTreeMap.size){
        array[i] = linkedTreeMap.get(fieldNames[i])
    }
    return array
}